src/common/pecan.hpp

```diff
@@ -0,0 +1,53 @@
+#ifndef PECAN_H
+#define PECAN_H
+#include "./ptypes.hpp"
+
+enum PCAN_ERR{
+    PACKET_TOO_BIG=-3,
+    NOT_RECEIVED=-2,
+    NOSPACE=-1,
+    SUCCESS = 0,
+    GEN_FAILURE = 1,
+};
+
+struct CANPacket {
+    int id;
+    char data[8] = {0};
+    char dataSize = 0;
+};
+
+// 
+struct PCANListenParamsCollection{
+    CANListenParam arr[1000];
```

Comment:
Change to a lower number, we're not expecting to have so much ids to listen for

Reply:
magic number

Reply:
done